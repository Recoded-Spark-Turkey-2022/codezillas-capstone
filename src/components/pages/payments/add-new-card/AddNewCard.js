/* eslint-disable jsx-a11y/label-has-associated-control */
import React, { useState, useEffect } from 'react';
import { doc, setDoc, getDoc } from 'firebase/firestore';
import { getAuth } from 'firebase/auth';
import { useFormik } from 'formik';
import swal from 'sweetalert';
import Card from 'react-credit-cards';
import Button from '../../../shared/button/Button';
import Select from '../select/Select';
import 'react-credit-cards/es/styles-compiled.css';
import db from '../../../../db/firebase.config';
import Footer from '../../../shared/footer/Footer';

function AddNewCard() {
  // Credit Card Numbers States
  const [CardNumber, setCardNumber] = useState();
  const [CardNumberFieldValidation, setCardNumberFieldValidation] =
    useState(true);
  const [MasterCard, setMasterCard] = useState(
    'border-white-cyan text-white-cyan'
  );
  const [VisaCard, setVisaCard] = useState('border-white-cyan text-white-cyan');

  // Credit Card Format
  const getMaskedValue = (value) =>
    Array.from(
      value
        .replaceAll(/\D/g, '')
        .matchAll(/(\d{0,4})(\d{0,4})(\d{0,4})(\d{0,4})/g)
    )[0]
      .slice(1, 5)
      .join(' ')
      .trim();

  // Credit Cards Handlers
  function handleCardNumber(e) {
    e.preventDefault();

    const number = e.target.value
      ? e.target.value
      : setCardNumberFieldValidation('invalid:border-red-500');
    const visa = /^4[0-9]/;
    const mastercard = /^5[1-5]/;
    const maestro = /^5[6-90]/;
    const invalid = /^[36]/;
    const validInput = /^[0-9]/;

    function handleMasterOrVisa(cardType) {
      if (cardType === 1) {
        setMasterCard('border-rose-600 text-rose-600 animate-pulse');
      } else if (cardType === 2) {
        setVisaCard('border-rose-600 text-rose-600 animate-pulse');
      } else {
        setMasterCard('border-white-cyan text-white-cyan');
        setVisaCard('border-white-cyan text-white-cyan');
        setCardNumber('');
      }
    }

    if (validInput.test(number)) {
      if (invalid.test(number)) {
        setCardNumberFieldValidation('peer invalid:border-red-500');
      } else {
        // eslint-disable-next-line no-lonely-if
        if (number.length > 1) {
          if (mastercard.test(number)) {
            handleMasterOrVisa(1);
            setCardNumber(getMaskedValue(number));
            setCardNumberFieldValidation('');
          } else if (visa.test(number)) {
            handleMasterOrVisa(2);
            setCardNumber(getMaskedValue(number));
            setCardNumberFieldValidation('');
          } else if (maestro.test(number)) {
            setCardNumberFieldValidation('peer invalid:border-red-500');
          } else {
            if (number.length <= 2) {
              setMasterCard('border-white-cyan text-white-cyan');
              setVisaCard('border-white-cyan text-white-cyan');
              setCardNumber(getMaskedValue(number));
            }
            setCardNumberFieldValidation('peer invalid:border-red-500');
          }
        } else {
          setMasterCard('border-white-cyan text-white-cyan');
          setVisaCard('border-white-cyan text-white-cyan');
          setCardNumber(getMaskedValue(number));
        }
      }
    } else {
      setCardNumberFieldValidation('peer invalid:border-red-500');
      setCardNumber('');
    }
  }

  // CVV States
  const [CVV, setCVV] = useState();

  // CVV Handlers
  function handleCVV(e) {
    e.preventDefault();
    const validInput = /^[0-9]/;
    if (validInput.test(e.target.value)) {
      setCVV(e.target.value);
    }

    if (e.target.value.length === 0) {
      setCVV(e.target.value);
    }
  }

  // Name States
  const [CardHolderName, setCardHolderName] = useState();

  // Name Handlers
  function handleCardHolderName(e) {
    e.preventDefault();
    const validInput = /[A-Za-z\s]$/;
    if (validInput.test(e.target.value)) {
      setCardHolderName(e.target.value);
    }

    if (e.target.value.length === 0) {
      setCardHolderName(e.target.value);
    }
  }

  // Expiry Date States
  const [ExpiryDate, setExpiryDate] = useState();

  // Expiry Date Handlers
  function handleExpiryDate(e) {
    e.preventDefault();
    const getMaskedDate = (value) =>
      value
        .replace(/[^0-9]/g, '')
        .replace(/^([2-9])$/g, '0$1')
        .replace(/^(1{1})([3-9]{1})$/g, '0$1/$2')
        .replace(/^0{1,}/g, '0')
        .replace(/^([0-1]{1}[0-9]{1})([0-9]{1,2}).*/g, '$1/$2');
    setExpiryDate(getMaskedDate(e.target.value));
  }

  // Zip Code States
  const [ZipCode, setZipCode] = useState();

  // Zip Code Handlers
  function handleZipCode(e) {
    e.preventDefault();
    setZipCode(e.target.value);
  }

  // Adress States
  const [Adress, setAdress] = useState();

  // Adress Handlers
  function handleAdress(e) {
    e.preventDefault();
    setAdress(e.target.value);
  }

  // Input Focus States
  const [InputFocus, setInputFocus] = useState('number');

  // Input Focus States
  function handleInputFocus(e) {
    if (e.target.name === 'cvc') {
      setInputFocus('cvc');
    } else {
      setInputFocus('number');
    }
  }

  // Select Countries options
  const [Countries, setCountries] = useState([
    { country_name: 'Loading....', country_phone_code: '' },
  ]);
  const [SelectedCountry, setSelectedCountry] = useState('');

  const [Cities, setCities] = useState([{ state_name: 'Loading....' }]);
  const [SelectedCity, setSelectedCity] = useState('');

  // Countries/Cities Access token Api
  const [AccessToken, setAccessToken] = useState('');

  // Get Cities based on the selected Country
  function fetchCities(city, AccessTokenA) {
    // console.log(AccessTokenA)
    setCities([{ state_name: 'Loading....' }]);
    fetch(`https://www.universal-tutorial.com/api/states/${city}`, {
      method: 'get',
      headers: new Headers({
        Accept: 'application/json',
        Authorization: `Bearer ${AccessTokenA || AccessToken}`,
      }),
    })
      .then((resul) => resul.json())
      .then((result) => setCities(result));
  }

  const formik = useFormik({
    initialValues: {
      cardNumber: '',
      expDate: '',
      CVV: '',
      country: '',
      city: '',
      zipCode: '',
      address: '',
    },
    onSubmit: async () => {
      const docRef = doc(db, 'user-details', getAuth().currentUser.email);
      const docSnap = await getDoc(docRef);
      const [
        cardNumber,
        expDate,
        country,
        city,
        cvvCode,
        nameOnCard,
        zipCode,
        fullAdress,
      ] = [
        CardNumber,
        ExpiryDate,
        SelectedCountry,
        SelectedCity,
        CVV,
        CardHolderName,
        ZipCode,
        Adress,
      ];

      if (!docSnap.exists()) {
        swal({
          title: 'Opps!',
          text: 'User Not Logged in',
          icon: 'error',
        });
      } else {
        setDoc(doc(db, 'user-details', getAuth().currentUser.email), {
          ...docSnap.data(),
          cards: [
            ...docSnap.data().cards,
            {
              cardNumber,
              expDate,
              country,
              city,
              cvvCode,
              nameOnCard,
              zipCode,
              fullAdress,
            },
          ],
        });
      }
    },
  });

  useEffect(() => {
    fetch('https://www.universal-tutorial.com/api/getaccesstoken', {
      method: 'get',
      headers: new Headers({
        Accept: 'application/json',
        'api-token': atob(
          'aDUyWXhaSHJGRkVYeVRzdUhfdlFWRm1jOEFvOEppY0lTbE9MVTUwcjk5RHRYQVZUY3NlUFZkVlJVTXUzTjkzQ0dETQ=='
        ),
        'user-email': 'nour.majed1234@gmail.com',
      }),
    })
      .then((resul) => resul.json())
      .then((result) => {
        fetchCities(undefined, result.auth_token);
        setAccessToken(result.auth_token);
        return result;
      })
      .then((jsonResult) =>
        fetch('https://www.universal-tutorial.com/api/countries/', {
          method: 'get',
          headers: new Headers({
            Accept: 'application/json',
            Authorization: `Bearer ${jsonResult.auth_token}`,
          }),
        })
          .then((resul) => resul.json())
          .then((result) => setCountries(result))
      );
  }, []);

  return (
    <>
      <form
        className="w-fit mx-10 my-8 lg:mx-16 lg:my-10  xl:mx-24 xl:my-16 grid gap-y-6 md:gap-y-12"
        onSubmit={formik.handleSubmit}
      >
        <div className="grid md:grid-cols-3">
          <span className="uppercase text-2xl lg:text-5xl col-span-3 mb-3">
            add card details
          </span>
          <span className="capitalize text-la lg:text-2xl col-span-3 row-start-auto w-fit text-[#00000080]">
            Please make sure all of the info you enter are the same as your card
            registration info.
          </span>
        </div>
        <div className="grid md:grid-cols-3 md:gap-x-6 lg:gap-x-9 gap-y-5  ">
          <div className="grid gap-6 ">
            <div className="h-20 grid content-center">
              <span className="text-zinc-400 ">Supported Card Type</span>
              <div className="grid grid-cols-2 ">
                <span className={`${MasterCard} border-[1px] text-center`}>
                  MasterCard
                </span>
                <span className={`${VisaCard} border-[1px] text-center`}>
                  Visa
                </span>
              </div>
            </div>
            <label className="text-zinc-400" htmlFor="cnumber">
              Card Number
              <input
                type="text"
                id="cnumber"
                name="number"
                className={`border w-full h-12 md:h-16 rounded-md pl-2 px-5 focus:outline-none focus:shadow-lg text-sm lg:text-base text-black  ${CardNumberFieldValidation}`}
                placeholder="4287  8874 9511 3263"
                maxLength="19"
                required
                value={CardNumber}
                onChange={(e) => handleCardNumber(e)}
                onFocus={(e) => handleInputFocus(e)}
              />
              <p className="mt-2 invisible peer-invalid:visible text-pink-600 text-sm">
                Please provide one of Supported cards
              </p>
            </label>
            <div className="grid grid-cols-2 gap-3">
              <label className="text-zinc-400" htmlFor="expiry">
                Expiry Date
                <input
                  type="text"
                  id="expiry"
                  name="expiry"
                  maxLength="5"
                  required
                  className="border w-full h-12 md:h-16 rounded-md pl-2 px-5 focus:outline-none focus:shadow-lg text-sm lg:text-base text-black"
                  placeholder="MM / YY"
                  value={ExpiryDate}
                  onChange={(e) => handleExpiryDate(e)}
                  onFocus={(e) => handleInputFocus(e)}
                />
              </label>

              <label className="text-zinc-400" htmlFor="cvv">
                CVV Code
                <input
                  id="cvv"
                  name="cvc"
                  maxLength="3"
                  required
                  className="border w-full h-12 md:h-16 rounded-md pl-2 px-5 focus:outline-none focus:shadow-lg text-sm lg:text-base text-black"
                  placeholder="***"
                  value={CVV}
                  onChange={(e) => handleCVV(e)}
                  onFocus={(e) => handleInputFocus(e)}
                />
              </label>
            </div>

            <label className="text-zinc-400" htmlFor="name">
              Name on Card
              <input
                type="text"
                id="name"
                name="name"
                pattern="[a-z A-Z-]+"
                required
                className="border w-full h-12 md:h-16 rounded-md pl-2 px-5 focus:outline-none focus:shadow-lg text-sm lg:text-base text-black"
                placeholder="Irene Ramos"
                value={CardHolderName}
                onChange={(e) => handleCardHolderName(e)}
                onFocus={(e) => handleInputFocus(e)}
              />
            </label>
          </div>
          <div className="grid gap-6 ">
            <label className="text-zinc-400" htmlFor="country">
              Country
              <Select
                id="country"
                AllClassess="border w-full h-12 md:h-16 rounded-md pl-2 px-5 focus:outline-none focus:shadow-lg text-sm lg:text-base text-black bg-white"
                options={Countries}
                selected={setSelectedCountry}
                defaultVal='Please Select Country'
                // eslint-disable-next-line react/jsx-no-bind
                func={fetchCities}
              />
            </label>

            <label className="text-zinc-400" htmlFor="zip">
              Zip Code
              <input
                type="text"
                id="zip"
                className="border w-full h-12 md:h-16 rounded-md pl-2 px-5 focus:outline-none focus:shadow-lg text-sm lg:text-base text-black"
                placeholder="17121-1300"
                minLength={1}
                required
                value={ZipCode}
                onChange={(e) => handleZipCode(e)}
              />
            </label>

            <label className="text-zinc-400" htmlFor="city">
              City
              <Select
                id="city"
                AllClassess="border w-full h-12 md:h-16 rounded-md pl-2 px-5 focus:outline-none focus:shadow-lg text-sm lg:text-base text-black bg-white"
                required
                options={Cities}
                defaultVal='Please Select City'
                selected={setSelectedCity}
              />
            </label>

            <label className="text-zinc-400" htmlFor="address">
              Adress
              <input
                type="text"
                id="address"
                className="border w-full h-12 md:h-16 rounded-md pl-2 px-5 focus:outline-none focus:shadow-lg text-sm lg:text-base text-black"
                required
                minLength={3}
                placeholder="509 Adele Mills Suite 833"
                value={Adress}
                onChange={(e) => handleAdress(e)}
              />
            </label>
          </div>
          <div className="grid place-content-center cursor-pointer">
            <Card
              cvc={CVV || ''}
              expiry={ExpiryDate || ''}
              name={CardHolderName || ''}
              number={CardNumber || ''}
              focused={InputFocus}
            />
          </div>
        </div>
        <div className="grid grid-cols-5">
          <Button
            type="submit"
            name="ADD CARD"
            classes="col-span-3 sm:col-span-2 md:col-span-1  py-2 md:py-3 px-5 md:h-14 hover:bg-cyan-200 focus:outline-none hover:scale-105 border border-cyan-400 bg-cyan-400 text-base sm:text-lg font-semibold rounded-md shadow-lg"
            // eslint-disable-next-line react/jsx-no-bind
            function={() => ''}
          />
        </div>
      </form>

      <Footer />
    </>
  );
}

export default AddNewCard;
