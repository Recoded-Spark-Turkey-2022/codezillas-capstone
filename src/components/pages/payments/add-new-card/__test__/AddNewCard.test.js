import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import AddNewCard from '../AddNewCard';
import { store } from '../../../../../app/store';

test('Navbar component renders correctly', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <BrowserRouter>
          <AddNewCard />
        </BrowserRouter>
      </Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
