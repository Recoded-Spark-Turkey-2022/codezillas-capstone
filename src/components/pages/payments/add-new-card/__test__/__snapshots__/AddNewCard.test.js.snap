// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component renders correctly 1`] = `
Array [
  <form
    className="w-fit mx-10 my-8 lg:mx-16 lg:my-10  xl:mx-24 xl:my-16 grid gap-y-6 md:gap-y-12"
    onSubmit={[Function]}
  >
    <div
      className="grid md:grid-cols-3"
    >
      <span
        className="uppercase text-2xl lg:text-5xl col-span-3 mb-3"
      >
        add card details
      </span>
      <span
        className="capitalize text-la lg:text-2xl col-span-3 row-start-auto w-fit text-[#00000080]"
      >
        Please make sure all of the info you enter are the same as your card registration info.
      </span>
    </div>
    <div
      className="grid md:grid-cols-3 md:gap-x-6 lg:gap-x-9 gap-y-5  "
    >
      <div
        className="grid gap-6 "
      >
        <div
          className="h-20 grid content-center"
        >
          <span
            className="text-zinc-400 "
          >
            Supported Card Type
          </span>
          <div
            className="grid grid-cols-2 "
          >
            <span
              className="border-white-cyan text-white-cyan border-[1px] text-center"
            >
              MasterCard
            </span>
            <span
              className="border-white-cyan text-white-cyan border-[1px] text-center"
            >
              Visa
            </span>
          </div>
        </div>
        <label
          className="text-zinc-400"
          htmlFor="cnumber"
        >
          Card Number
          <input
            className="border w-full h-12 md:h-16 rounded-md pl-2 px-5 focus:outline-none focus:shadow-lg text-sm lg:text-base text-black  true"
            id="cnumber"
            maxLength="19"
            name="number"
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="4287  8874 9511 3263"
            required={true}
            type="text"
          />
          <p
            className="mt-2 invisible peer-invalid:visible text-pink-600 text-sm"
          >
            Please provide one of Supported cards
          </p>
        </label>
        <div
          className="grid grid-cols-2 gap-3"
        >
          <label
            className="text-zinc-400"
            htmlFor="expiry"
          >
            Expiry Date
            <input
              className="border w-full h-12 md:h-16 rounded-md pl-2 px-5 focus:outline-none focus:shadow-lg text-sm lg:text-base text-black"
              id="expiry"
              maxLength="5"
              name="expiry"
              onChange={[Function]}
              onFocus={[Function]}
              placeholder="MM / YY"
              required={true}
              type="text"
            />
          </label>
          <label
            className="text-zinc-400"
            htmlFor="cvv"
          >
            CVV Code
            <input
              className="border w-full h-12 md:h-16 rounded-md pl-2 px-5 focus:outline-none focus:shadow-lg text-sm lg:text-base text-black"
              id="cvv"
              maxLength="3"
              name="cvc"
              onChange={[Function]}
              onFocus={[Function]}
              placeholder="***"
              required={true}
            />
          </label>
        </div>
        <label
          className="text-zinc-400"
          htmlFor="name"
        >
          Name on Card
          <input
            className="border w-full h-12 md:h-16 rounded-md pl-2 px-5 focus:outline-none focus:shadow-lg text-sm lg:text-base text-black"
            id="name"
            name="name"
            onChange={[Function]}
            onFocus={[Function]}
            pattern="[a-z A-Z-]+"
            placeholder="Irene Ramos"
            required={true}
            type="text"
          />
        </label>
      </div>
      <div
        className="grid gap-6 "
      >
        <label
          className="text-zinc-400"
          htmlFor="country"
        >
          Country
          <select
            className="border w-full h-12 md:h-16 rounded-md pl-2 px-5 focus:outline-none focus:shadow-lg text-sm lg:text-base text-black bg-white"
            onChange={[Function]}
          >
            <option
              selected="selected"
            >
              Please Select Country
            </option>
            <option
              value="Loading...."
            >
              Loading....
            </option>
          </select>
        </label>
        <label
          className="text-zinc-400"
          htmlFor="zip"
        >
          Zip Code
          <input
            className="border w-full h-12 md:h-16 rounded-md pl-2 px-5 focus:outline-none focus:shadow-lg text-sm lg:text-base text-black"
            id="zip"
            minLength={1}
            onChange={[Function]}
            placeholder="17121-1300"
            required={true}
            type="text"
          />
        </label>
        <label
          className="text-zinc-400"
          htmlFor="city"
        >
          City
          <select
            className="border w-full h-12 md:h-16 rounded-md pl-2 px-5 focus:outline-none focus:shadow-lg text-sm lg:text-base text-black bg-white"
            onChange={[Function]}
          >
            <option
              selected="selected"
            >
              Please Select Country
            </option>
            <option
              value="Loading...."
            >
              Loading....
            </option>
          </select>
        </label>
        <label
          className="text-zinc-400"
          htmlFor="address"
        >
          Adress
          <input
            className="border w-full h-12 md:h-16 rounded-md pl-2 px-5 focus:outline-none focus:shadow-lg text-sm lg:text-base text-black"
            id="address"
            minLength={3}
            onChange={[Function]}
            placeholder="509 Adele Mills Suite 833"
            required={true}
            type="text"
          />
        </label>
      </div>
      <div
        className="grid place-content-center cursor-pointer"
      >
        <div
          className="rccs"
        >
          <div
            className="rccs__card rccs__card--unknown"
          >
            <div
              className="rccs__card--front"
            >
              <div
                className="rccs__card__background"
              />
              <div
                className="rccs__issuer"
              />
              <div
                className="rccs__cvc__front"
              />
              <div
                className="rccs__number  rccs--focused"
              >
                •••• •••• •••• ••••
              </div>
              <div
                className="rccs__name"
              >
                YOUR NAME HERE
              </div>
              <div
                className="rccs__expiry"
              >
                <div
                  className="rccs__expiry__valid"
                >
                  valid thru
                </div>
                <div
                  className="rccs__expiry__value"
                >
                  ••/••
                </div>
              </div>
              <div
                className="rccs__chip"
              />
            </div>
            <div
              className="rccs__card--back"
            >
              <div
                className="rccs__card__background"
              />
              <div
                className="rccs__stripe"
              />
              <div
                className="rccs__signature"
              />
              <div
                className="rccs__cvc"
              />
              <div
                className="rccs__issuer"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="grid grid-cols-5"
    >
      <input
        className="py-2 px-5 w-10/12 md:w-8/12 lg:w-5/12 hover:bg-cyan-200 bg-cyan-400 font-medium rounded text-sm md:text-lg"
        onClick={[Function]}
        type="button"
        value="ADD CARD"
      />
    </div>
  </form>,
  <div
    className="w-full h-60 md:h-44 bg-amber-200 flex justify-center"
  >
    <div
      className="w-full md:w-10/12 h-full flex flex-col md:flex-row justify-between items-center"
    >
      <div
        className="h-5/6 w-5/6 md:w-4/12 flex flex-col items-center md:items-start justify-around"
      >
        <div
          className="flex flex-col items-center md:items-start gap-2 w-full "
        >
          <h2
            className="text-2xl font-semibold"
          >
            Subscribe
          </h2>
          <p
            className="text-gray-500 "
          >
            We\`ll never to spam you or share your email
          </p>
        </div>
        <div
          className="flex justify-start items-center"
        >
          <input
            className="border border-gray-400 rounded-l-md h-10 pl-2 focus:outline-none "
            onChange={[Function]}
            placeholder="Enter your e-mail"
            type="email"
            value=""
          />
          <button
            className="h-10  bg-cyan-400 pl-1 pr-2 rounded-r-md border-y border-r border-gray-400 hover:bg-cyan-300 text-xl"
            onClick={[Function]}
            type="button"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 448 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M446.7 98.6l-67.6 318.8c-5.1 22.5-18.4 28.1-37.3 17.5l-103-75.9-49.7 47.8c-5.5 5.5-10.1 10.1-20.7 10.1l7.4-104.9 190.9-172.5c8.3-7.4-1.8-11.5-12.9-4.1L117.8 284 16.2 252.2c-22.1-6.9-22.5-22.1 4.6-32.7L418.2 66.4c18.4-6.9 34.5 4.1 28.5 32.2z"
              />
            </svg>
          </button>
        </div>
      </div>
      <div
        className="h-5/6 w-full md:w-4/12 flex flex-col justify-evenly items-center"
      >
        <div
          className="w-full flex justify-between items-center gap-2"
        >
          <a
            className="text-gray-500 hover:scale-110 hover:underline hover:underline-offset-4 text-lg"
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
          <a
            className="text-gray-500 hover:scale-110 hover:underline hover:underline-offset-4 text-lg"
            href="/blogs"
            onClick={[Function]}
          >
            Blogs
          </a>
          <a
            className="text-gray-500 hover:scale-110 hover:underline hover:underline-offset-4 text-lg"
            href="/about"
            onClick={[Function]}
          >
            About
          </a>
          <a
            className="text-gray-500 hover:scale-110 hover:underline hover:underline-offset-4 text-lg"
            href="/contact"
            onClick={[Function]}
          >
            Contact
          </a>
        </div>
        <div
          className="w-6/12 flex justify-around items-center"
        >
          <button
            alt="Twitter"
            className="text-3xl hover:text-cyan-400"
            type="button"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M459.37 151.716c.325 4.548.325 9.097.325 13.645 0 138.72-105.583 298.558-298.558 298.558-59.452 0-114.68-17.219-161.137-47.106 8.447.974 16.568 1.299 25.34 1.299 49.055 0 94.213-16.568 130.274-44.832-46.132-.975-84.792-31.188-98.112-72.772 6.498.974 12.995 1.624 19.818 1.624 9.421 0 18.843-1.3 27.614-3.573-48.081-9.747-84.143-51.98-84.143-102.985v-1.299c13.969 7.797 30.214 12.67 47.431 13.319-28.264-18.843-46.781-51.005-46.781-87.391 0-19.492 5.197-37.36 14.294-52.954 51.655 63.675 129.3 105.258 216.365 109.807-1.624-7.797-2.599-15.918-2.599-24.04 0-57.828 46.782-104.934 104.934-104.934 30.213 0 57.502 12.67 76.67 33.137 23.715-4.548 46.456-13.32 66.599-25.34-7.798 24.366-24.366 44.833-46.132 57.827 21.117-2.273 41.584-8.122 60.426-16.243-14.292 20.791-32.161 39.308-52.628 54.253z"
              />
            </svg>
          </button>
          <button
            alt="Facebook"
            className="text-3xl hover:text-blue-800"
            type="button"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M504 256C504 119 393 8 256 8S8 119 8 256c0 123.78 90.69 226.38 209.25 245V327.69h-63V256h63v-54.64c0-62.15 37-96.48 93.67-96.48 27.14 0 55.52 4.84 55.52 4.84v61h-31.28c-30.8 0-40.41 19.12-40.41 38.73V256h68.78l-11 71.69h-57.78V501C413.31 482.38 504 379.78 504 256z"
              />
            </svg>
          </button>
          <button
            alt="Google"
            className="text-3xl hover:text-white"
            type="button"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M5.26620003,9.76452941 C6.19878754,6.93863203 8.85444915,4.90909091 12,4.90909091 C13.6909091,4.90909091 15.2181818,5.50909091 16.4181818,6.49090909 L19.9090909,3 C17.7818182,1.14545455 15.0545455,0 12,0 C7.27006974,0 3.1977497,2.69829785 1.23999023,6.65002441 L5.26620003,9.76452941 Z"
              />
              <path
                d="M16.0407269,18.0125889 C14.9509167,18.7163016 13.5660892,19.0909091 12,19.0909091 C8.86648613,19.0909091 6.21911939,17.076871 5.27698177,14.2678769 L1.23746264,17.3349879 C3.19279051,21.2936293 7.26500293,24 12,24 C14.9328362,24 17.7353462,22.9573905 19.834192,20.9995801 L16.0407269,18.0125889 Z"
              />
              <path
                d="M19.834192,20.9995801 C22.0291676,18.9520994 23.4545455,15.903663 23.4545455,12 C23.4545455,11.2909091 23.3454545,10.5272727 23.1818182,9.81818182 L12,9.81818182 L12,14.4545455 L18.4363636,14.4545455 C18.1187732,16.013626 17.2662994,17.2212117 16.0407269,18.0125889 L19.834192,20.9995801 Z"
              />
              <path
                d="M5.27698177,14.2678769 C5.03832634,13.556323 4.90909091,12.7937589 4.90909091,12 C4.90909091,11.2182781 5.03443647,10.4668121 5.26620003,9.76452941 L1.23999023,6.65002441 C0.43658717,8.26043162 0,10.0753848 0,12 C0,13.9195484 0.444780743,15.7301709 1.23746264,17.3349879 L5.27698177,14.2678769 Z"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </div>,
]
`;
